src/api/Base.js

```diff
@@ -219,10 +219,24 @@ class Base {
      * @param {Function} successCallback - The success callback
      * @param {Function} errorCallback - The error callback
      * @param {Object} params request params
+     * @param {string} url - API url
+     * @returns {Promise}
      */
-    get(id: string, successCallback: Function, errorCallback: Function, params?: Object): void {
-        const url = this.getUrl(id);
-        this.makeRequest(HTTP_GET, id, url, successCallback, errorCallback, params);
+    get({
```

Comment:
Would it be worth updating `post`, `put`, and `delete` to work with a similar fashion of arguments?

Reply:
I can do that in a follow up commit 👍

Reply:
Done in #407